src/snapshot.c

```diff
@@ -568,19 +551,74 @@ int raftLoadSnapshot(raft_server_t *raft, void *user_data, raft_term_t term, raf
     raft_end_load_snapshot(rr->raft);
 
     /* Restart the log where the snapshot ends */
-    if (rr->log) {
-        fsyncThreadWaitUntilCompleted(&rr->fsyncThread);
-        RaftLogClose(rr->log);
-        rr->log = RaftLogCreate(rr->config.log_filename,
-                                rr->snapshot_info.dbid,
-                                rr->snapshot_info.last_applied_term,
-                                rr->snapshot_info.last_applied_idx,
-                                &rr->config);
-        EntryCacheDeleteHead(rr->logcache, raft_get_snapshot_last_idx(rr->raft) + 1);
-    }
+    fsyncThreadWaitUntilCompleted(&rr->fsyncThread);
+    RaftLogClose(rr->log);
+    rr->log = RaftLogCreate(rr->config.log_filename,
+                            rr->snapshot_info.dbid,
+                            rr->snapshot_info.last_applied_term,
+                            rr->snapshot_info.last_applied_idx,
+                            &rr->config);
+
+    EntryCacheDeleteHead(rr->logcache, raft_get_snapshot_last_idx(rr->raft) + 1);
 
     createOutgoingSnapshotMmap(rr);
     rr->snapshots_received++;
+    rr->state = REDIS_RAFT_UP;
+
+    rr->snapshot_load = (SnapshotLoad){
+        .pending = false,
+        .term = -1,
+        .index = -1,
+    };
+}
+
+/* After a snapshot is received, load it into the Raft library:
+ * 1. Replace received snapshot file with the current one.
```

Comment:
Since `raftLoadSnapshot` is not loading any more the snapshot, but only triggers a loading state of RR, I suggest changing the documentation here accordingly  + adding some documentation in `loadPendingSnapshot`

Reply:
ok added some docs